src/tests/ftest/pool/bad_evict.py

```diff
@@ -50,39 +50,38 @@ def test_evict(self):
 
         saveduuid = None
 
-        try:
-            # initialize a python pool object then create the underlying
-            # daos storage
-            self.add_pool(connect=False)
-
-            # trash the UUID value in various ways
-            if excludeuuid is None:
-                saveduuid = (ctypes.c_ubyte * 16)(0)
-                for item in range(0, len(saveduuid)):
-                    saveduuid[item] = self.pool.pool.uuid[item]
-                self.pool.pool.uuid[0:] = \
-                    [0 for item in range(0, len(self.pool.pool.uuid))]
-            elif excludeuuid == 'JUNK':
-                saveduuid = (ctypes.c_ubyte * 16)(0)
-                for item in range(0, len(saveduuid)):
-                    saveduuid[item] = self.pool.pool.uuid[item]
-                self.pool.pool.uuid[4] = 244
-
-            # evict the pool
-            self.get_dmg_command().pool_evict(self.pool.pool.get_uuid_str())
-
-            if expected_result in ['FAIL']:
-                self.fail("Test was expected to fail but it passed.\n")
-
-        except CommandFailure as excep:
-            self.log.error(str(excep))
-            self.log.error(traceback.format_exc())
-            if expected_result in ['PASS']:
-                self.fail("Test was expected to pass but it failed.\n")
-        finally:
-            if self.pool is not None:
-                # if the test trashed some pool parameter, put it back the
-                # way it was
-                if saveduuid is not None:
-                    self.pool.pool.uuid = saveduuid
-                self.pool.destroy()
+        self.add_pool(connect=False)
+        original_test_pool_uuid = self.pool.uuid
+
+        # Make dmg call and poolevict() not fail.
+        self.pool.dmg.exit_status_exception = False
+
+        # trash the UUID value in various ways
+        if excludeuuid is None:
+            saveduuid = (ctypes.c_ubyte * 16)(0)
+            for item in range(0, len(saveduuid)):
+                saveduuid[item] = self.pool.pool.uuid[item]
+            self.pool.pool.uuid[0:] = \
+                [0 for item in range(0, len(self.pool.pool.uuid))]
+        elif excludeuuid == 'JUNK':
+            saveduuid = (ctypes.c_ubyte * 16)(0)
+            for item in range(0, len(saveduuid)):
```

Comment:
(pylint-consider-using-enumerate) Consider using enumerate instead of iterating with range and len